tango/__main__.py

```diff
@@ -774,12 +775,14 @@ def log_and_execute_run():
                 raise CliRunError
 
     if called_by_executor:
+        assert step_names
+
         from tango.common.aliases import EnvVarNames
 
         # We set this environment variable so that any steps that contain multiprocessing
         # and call `initialize_worker_logging` also log the messages with the `step_name` prefix.
-        os.environ[EnvVarNames.LOGGING_PREFIX.value] = f"step {step_name}"
-        initialize_prefix_logging(prefix=f"step {step_name}", main_process=False)
+        os.environ[EnvVarNames.LOGGING_PREFIX.value] = f"step {step_names[0]}"
+        initialize_prefix_logging(prefix=f"step {step_names[0]}", main_process=False)
```

Comment:
Does this make sense? All the log messages have the prefix of the first step? That feels like it'll lead to confusion.

Why do we even have to say the step name in the prefix for the logs? Is it OK if we only do that when there is only one step, and otherwise we say "multiple steps", or no prefix at all?

Reply:
This only happens when `called_by_executor` is `True`, in which case `len(step_names) == 1`. 2ef025b